contracts/exchange/contracts/src/MixinMatchOrders.sol

```diff
@@ -263,6 +259,446 @@ contract MixinMatchOrders is
+            matchedFillResults.right.takerAssetFilledAmount = rightTakerAssetAmountRemaining;
+        }
+
+        // Calculate amount given to taker
+        matchedFillResults.profitInLeftMakerAsset = _safeSub(
+            matchedFillResults.left.makerAssetFilledAmount,
+            matchedFillResults.right.takerAssetFilledAmount
+        );
+    }
+
+    /// @dev Calculates part of the matched fill results for a given situation using the maximal fill order matching
+    ///      strategy.
+    /// @param matchedFillResults The MatchedFillResults struct to update with fill result calculations.
+    /// @param leftOrder The left order in the order matching situation.
+    /// @param rightOrder The right order in the order matching situation.
+    /// @param leftMakerAssetAmountRemaining The amount of the left order maker asset that can still be filled.
+    /// @param leftTakerAssetAmountRemaining The amount of the left order taker asset that can still be filled.
+    /// @param rightMakerAssetAmountRemaining The amount of the right order maker asset that can still be filled.
+    /// @param rightTakerAssetAmountRemaining The amount of the right order taker asset that can still be filled.
+    function _calculateMatchedFillResultsWithMaximalFill(
+        MatchedFillResults memory matchedFillResults,
+        LibOrder.Order memory leftOrder,
+        LibOrder.Order memory rightOrder,
+        uint256 leftMakerAssetAmountRemaining,
+        uint256 leftTakerAssetAmountRemaining,
+        uint256 rightMakerAssetAmountRemaining,
+        uint256 rightTakerAssetAmountRemaining
+    )
+        internal
+        pure
+    {
+        bool doesLeftMakerAssetProfitExist;
+        bool doesRightMakerAssetProfitExist;
+
+        // Calculate the maximum fill results for the maker and taker assets. At least one of the orders will be fully filled.
+        //
+        // The maximum that the left maker can possibly buy is the amount that the right order can sell.
+        // The maximum that the right maker can possibly buy is the amount that the left order can sell.
+        //
+        // If the left order is fully filled, profit will be paid out in the left maker asset. If the right order is fully filled,
+        // the profit will be out in the right maker asset.
+        //
+        // There are three cases to consider:
+        // Case 1.
+        //   If the left maker can buy more than or equal to the right maker can sell, then only the right order is fully filled.
+        // Case 2.
+        //   If the right maker can buy more than or equal to the left maker can sell, then only the right order is fully filled.
+        // Case 3.
+        //   If the right maker can sell the max of what the left maker can buy and the left maker can sell the max of
+        //   what the right maker can buy, then both orders are fully filled.
+        if (leftTakerAssetAmountRemaining >= rightMakerAssetAmountRemaining) {
```

Comment:
I believe we want to make these `>` opposed to `>=`, since both orders will be fully filled in the `==` case. This would be aligned with the regular `matchOrders` and would eliminate unnecessary rounding in the `==` case.

Reply:
Agreed, it'll be more accurate.